### DISCUSSION TITLE
How To Code CSS in less characters to win CSS Batle?

### DISCUSSION THREAD

**Discussion by NaviTalib** [cssbattle/forum#344] [General]
I just participated in CSS Battle, and Complete my first challenge within  200+ characters but I realize that another person did that only in 60 characters, then I also try to reduce my code and made it within 113 characters.
but I need to make it shorter to serve in the top 100.

kindly help me with it.

**Comment by chinchang** [comment]
Let's start here -> https://cssbattle.dev/tips

**Reply by NaviTalib** [reply]
OK,
this is my code of simple circle:

<p><style>p{width:200;height:200;background:#d86f45;border-radius:50%}*{background:#f5d6b4;margin:0

is this possible to make it in less than 99 characters?

**Reply by verberg** [reply]
For example radial-gradients give you a better solution with less chars:
<style>* *{background:radial-gradient(1q,#D86F45,100px,#F5D6B4)-100px -50px

**Reply by NaviTalib** [reply]
@verberg that code is working giving 100% accuracy within 75 chars.
Thank you.

**Reply by verberg** [reply]
You can improve it by replacing the px-units with q or vw.

**Comment by SelenIT** [comment]
IMO, the 1st rule of CSS code golfing is to change the mindset from "How to style _this element_ to make this shape?" to "What CSS _features_ can be used to make this shape?"

If it's a circle, can it be a dot of the dotted border? If there are several concentric circles in the center of the viewport, can they all be drawn with just one `radial-gradient`? If there is a regular pattern (like in [Target #18](https://cssbattle.dev/play/18)), can it be a repeating background of `body`? Can the parts of the shape be the layers of one element's `box-shadow` (like in [Target #7](https://cssbattle.dev/play/7))? Can two (or even four) parallel lines be done with just one `border: double` declaration?

While properties like `position`, `width`/`height`, `top`/`left`, etc. are pretty common to CSS drawing as well as making each part of the image with a dedicated element, this approach is usually too "expensive" for code golfing and such properties almost never occur in top solutions. [It's proven possible](https://github.com/MSanbira/css_battle_only_gradients) to solve any CSSBattle targets with just `background` of the root element, and many top solutions actually have nothing but default `html` and `body` with their backgrounds or shadows, sometimes with `margin` for sizing and placing them in the same time. Adding an extra element becomes more like a last resort, most likely either to reduce repetition in CSS or to make a totally different shape hardly doable with existing markup.
